src/components/molecules/search/Search.jsx

```diff
@@ -0,0 +1,191 @@
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from 'react';
+import { useLocation, useNavigate, useParams } from 'react-router-dom';
+import { useTracking } from 'react-tracking';
+
+import './Search.scss';
+
+import { Autocomplete, Radio } from '../../atomic';
+import {
+	searchMatchType,
+	searchMatchTypeAnalyticsMap,
+} from '../../../constants';
+import { useAppPaths, useCustomQuery } from '../../../hooks';
+import { getAutoSuggestResults } from '../../../services/api/actions';
+import {
+	emboldenSubstring,
+	getKeyValueFromObject,
+	getKeyValueFromQueryString,
+	matchItemToTerm,
+} from '../../../utils';
+
+const Search = ({ autoSuggestLimit = 10 }) => {
+	const params = useParams();
+	const location = useLocation();
+	const { searchText: urlParamSearchText } = params;
+	const { search } = location;
+	const tracking = useTracking();
+
+	// Set matchType to value retrieved from url if it exits and default to "Begins" if not
+	const matchType =
+		search && getKeyValueFromQueryString('searchMode', search) !== null
+			? getKeyValueFromQueryString('searchMode', search)
+			: searchMatchType.beginsWith;
+	// Set default selected option for search match type
+	const [selectedOption, updateSelectedOption] = useState(matchType);
+	// Set default search text to value retrieved from url or set to empty string if not
+	const [searchText, updateSearchText] = useState(
+		urlParamSearchText ? decodeURIComponent(urlParamSearchText) : ''
+	);
+	const [shouldFetchAutoSuggest, setFetchAutoSuggest] = useState(false);
+	const navigate = useNavigate();
+	const { SearchPath } = useAppPaths();
+	const autoSuggest = useCustomQuery(
+		getAutoSuggestResults(searchText, selectedOption, autoSuggestLimit),
+		shouldFetchAutoSuggest
+	);
+
+	useEffect(() => {
+		// Set selected option value if url parameters change
+		updateSelectedOption(matchType);
+	}, [matchType]);
+
+	const trackSubmit = () => {
+		const searchType =
+			selectedOption &&
+			getKeyValueFromObject(selectedOption, searchMatchTypeAnalyticsMap)
+				? getKeyValueFromObject(selectedOption, searchMatchTypeAnalyticsMap)
+				: searchMatchTypeAnalyticsMap[searchMatchType.beginsWith];
+		tracking.trackEvent({
+			type: 'Other',
+			event: 'DrugDictionaryApp:Other:KeywordSearch',
+			linkName: 'DrugDictionarySearch',
+			searchTerm: searchText,
+			searchType: searchType,
```

Comment:
The event is missing one parameter :  | analyticsName     | Drug |

Reply:
Added

Reply:
Don't most of the click events and page loads have analyticsName? If so it should be wired up at the analytics provider level. See https://github.com/NCIOCPL/glossary-app/blob/develop/src/index.js#L85